src/Solvers/ihs.jl

```diff
@@ -0,0 +1,234 @@
+    log::Logger
+    compressor::Compressor
+    error::SolverError
+    function IHS(alpha, log, compressor, error)
+        if typeof(compressor.cardinality) != Left
+            @warn "Compressor has cardinality `Right` but IHS compresses from the `Left`."
+        end 
+
+        new(alpha, log, compressor, error)
+    end
+
+end
+
+function IHS(;
+    compressor::Compressor = SparseSign(cardinality = Left()),
+    log::Logger = BasicLogger(),
+    error::SolverError = FullResidual(),
+    alpha::Float64 = 1.0
+)
+    return IHS(
+        alpha, 
+        log,
+        compressor,
+        error
+    )
+end
+
+"""
+    IHSRecipe{
+        Type<:Number, 
+        LR<:LoggerRecipe,
+        CR<:CompressorRecipe,
+        ER<:ErrorRecipe,
+        M<:AbstractArray, 
+        MV<:SubArray, 
+        V<:AbstractVector
+    } <: SolverRecip
+
+A mutable structure containing all information relevant to the Iterative Hessian Sketch 
+solver. It is formed by calling the function `complete_solver` on a `IHS` solver, which 
+includes all the user controlled parameters, the linear system `A`, and the constant 
+vector `b`.
+
+# Fields
+- `compressor::CompressorRecipe`, a technique for compressing the matrix ``A``.
+- `logger::LoggerRecipe`, a technique for logging the progress of the solver.
+- `error::SolverErrorRecipe`, a technique for estimating the progress of the solver.
+- `compressed_mat::AbstractMatrix`, a buffer for storing the compressed matrix.
+- `mat_view::SubArray`, a container for storing a view of the compressed matrix buffer.
+- `residual_vec::AbstractVector`, a vector that contains the residual of the linear system 
+    ``Ax-b``.
+- `gradient_vec::AbstractVector`, a vector that contains the gradient of the least squares 
+    problem, ``A^\\top(b-Ax)``.
+- `buffer_vec::AbstractVector`, a buffer vector for storing intermediate linear system solves.
+- `solution_vec::AbstractVector`, a vector storing the current IHS solution.
+- `R::UpperTriangular`, a container for storing the upper triangular portion of the R 
+    factor from a QR factorization of `mat_view`. This is used to solve the IHS sub-problem.
+"""
+mutable struct IHSRecipe{
+    Type<:Number, 
+    LR<:LoggerRecipe,
+    CR<:CompressorRecipe,
+    ER<:SolverErrorRecipe,
+    M<:AbstractArray, 
+    MV<:SubArray, 
+    V<:AbstractVector
+} <: SolverRecipe
+    log::LR
+    compressor::CR
+    error::ER
+    alpha::Float64
+    compressed_mat::M
+    mat_view::MV
+    residual_vec::V
+    gradient_vec::V
+    buffer_vec::V
+    solution_vec::V
+    R::UpperTriangular{Type, M}
+end
+
+function complete_solver(
+    ingredients::IHS, 
+    x::AbstractVector, 
+    A::AbstractMatrix, 
+    b::AbstractVector
+)
+    compressor = complete_compressor(ingredients.compressor, x, A, b)
+    logger = complete_logger(ingredients.log) 
+    error = complete_error(ingredients.error, ingredients, A, b)
+    sample_size::Int64 = compressor.n_rows
+    rows_a, cols_a = size(A)
```

Comment:
Add check if `rows_a <= cols_a` and throw an error. Make sure to test for this error